src/sinks/datadog_archives.rs

```diff
@@ -562,41 +570,43 @@ struct DatadogGcsRequestBuilder {
 }
 
 impl RequestBuilder<(String, Vec<Event>)> for DatadogGcsRequestBuilder {
-    type Metadata = GcsMetadata;
+    type Metadata = (String, usize, EventFinalizers, usize);
     type Events = Vec<Event>;
     type Payload = Bytes;
     type Request = GcsRequest;
     type Encoder = DatadogArchivesEncoding;
     type Error = io::Error;
 
     fn split_input(&self, input: (String, Vec<Event>)) -> (Self::Metadata, Self::Events) {
-        let (key, mut events) = input;
+        let (partition_key, mut events) = input;
+        let event_count = events.len();
+        let event_byte_size = events.size_of();
         let finalizers = events.take_finalizers();
-        let metadata = GcsMetadata {
-            key,
-            count: events.len(),
-            byte_size: events.size_of(),
-            finalizers,
-        };
 
-        (metadata, events)
+        (
+            (partition_key, event_count, finalizers, event_byte_size),
+            events,
+        )
```

Comment:
This looks like the new metadata has all the same data as the old. What is the motivation for avoiding `GcsMetadata` here?

Reply:
My comment here explains the rationale: https://github.com/vectordotdev/vector/pull/12755#discussion_r881134240